### PR TITLE
fix(inference): raise qwen3-coder-30b-a3b context window to 16384

### PR DESCRIPTION
**Pull request by rustnew**
## 1. Summary

This PR changes:

- `charts/inference/values.yaml`: `qwen3-coder-30b-a3b` `contextSize` 12288 → 16384 (renders `--max-model-len 16384`)
- `charts/ai-models/values.yaml`: `qwen3-coder-30b-a3b-local` `contextLength` 12288 → 16384 (catalog)
- `charts/librechat-opencode-wellknown/values.yaml`: window comment 12288 → 16384
- `docs/adr/0118-fp8-kv-cache-fleet-default.md`: note that the one enabled vLLM model opts out of the fp8 KV default

It solves:

- Production `ContextOverflowError` (HTTP 400): a 8193-token prompt + 4096 requested output = 12289 > 12288 → vLLM refuses, because `--max-model-len` is the **total** window (input + output) and `maxOutputTokens` is only an advertisement, not a server-enforced cap.

## 2. Intent

The intent of this PR is:

> 16384 is the smallest window where the advertised `contextLength` (12288) + `maxOutputTokens` (4096) both fit, so a client honouring the catalog can never exceed the server window. This is the same failure as the 8193 > 8192 refusal fixed in #919, one step higher — the documented agentic-coding scenario (system prompt + tools + file context > 8K on turn one) was never actually covered at 12288. Follows #919 and #917.

---

### In Scope

- Raise the server window and the catalog `contextLength` to 16384 for `qwen3-coder-30b-a3b` only
- Update the wellknown comment and ADR-0118 blast-radius note

### Out of Scope

- Measuring real VRAM at the load gate (tracked as a load-gate follow-up)
- Other models' context windows
- Client-side `maxOutputTokens` tuning

---

## 4. Verification

I verified this change by:

- [x] Running automated tests
- [x] Testing error cases
- [ ] Checking logs
- [ ] Checking metrics
- [ ] Testing permissions/security behavior
- [ ] Testing rollback or failure behavior, if relevant

Commands run:

```bash
helm lint charts/inference charts/ai-models charts/librechat-opencode-wellknown
helm template test charts/inference   # confirms --max-model-len 16384, --gpu-memory-utilization 0.95, --max-num-seqs 1, --kv-cache-dtype auto
helm template test charts/ai-models   # confirms contextLength 16384 / maxOutputTokens 4096
bash tools/check-model-catalogs.sh
```

Results:

```text
3 chart(s) linted, 0 chart(s) failed
ALL CHARTS PASS   (full repo audit, 55 charts, CI-matched logic)
check-model-catalogs: OK (2 served on the GPU fleet, 2 federated cluster-local)
Arithmetic: 12288 + 4096 = 16384 = 16384 → no client honouring the catalog can exceed the window
KV @16384 = ~1.50 GiB → ~1.83 GiB left for activations (budget 0.95: 18.99 GiB − 15.66 GiB weights)
```

---

## 5. Screenshots / Evidence

Add evidence here:

* Metrics: [link]
* Logs: production error this PR fixes — `ContextOverflowError` `(parameter=input_tokens, value=8193)`, "maximum context length is 12288", total 12289 > 12288 → HTTP 400

---

## 6. Risk Assessment

Risk level:

* [ ] Low
* [x] Medium
* [ ] High

Potential risks:

- The ~1.83 GiB activation headroom at 16384 is arithmetic, not measured — if vLLM aborts at startup ("no memory for KV blocks"), the window must drop back to 12288 and the catalog advertise 8192 input

Mitigation:

- Re-verify actual VRAM at the load gate after deploy; the parallel: 1 and 16-bit KV cache (no fp8) settings keep the footprint minimal

---

## 7. AI Usage Declaration

AI was used for:

- [x] Understanding existing code
- [x] Reviewing the diff
- [x] Drafting documentation
- [ ] Generating code
- [ ] Refactoring
- [ ] Generating tests
- [ ] Not used

Human verification:

- [x] I understand every meaningful change in this PR
- [x] I checked generated code manually
- [x] I checked generated tests manually
- [x] I removed unsupported AI assumptions
- [x] I accept responsibility for this PR

---

## 8. Reviewer Focus

Please focus your review on:

- [x] Correctness
- [ ] Architecture
- [x] Security
- [ ] Performance
- [ ] Tests
- [x] Maintainability
- [x] Product intent
- [x] Edge cases

### CODE REVIEW

File: charts/ai-models/values.yaml [hunk 1]
```diff
@@ -798,7 +805,7 @@ models:
       connectionIdleTimeout: 1h
     info:
       displayName: "Qwen3-Coder-30B (self-hosted)"
-      contextLength: 12288           # = --max-model-len (vLLM: the per-request window)
+      contextLength: 16384           # = --max-model-len (vLLM: the per-request window)
```
[the remark this answers is not in the document]
Comment (rustnew) [reply charts/ai-models/values.yaml]: Confirmed — the P1 is valid and it has been fixed in `47387251`.

**Finding:** I had set `contextLength: 16384` in ai-models, equal to the serving window (16384). Under the change's own model (input + additive maxOutputTokens must fit the window), a client honouring the catalog could send 16384 input + 4096 output = 20480 > 16384 → the same vLLM 400 this PR removes, one step higher.

**Fix (`47387251`):** the catalog now advertises the INPUT window, not the serving window:
- `charts/ai-models/values.yaml`: `contextLength: 16384` → `12288` (`# = advertised INPUT window (clients add maxOutputTokens on top; serving window is 16384)`)
- `charts/librechat-opencode-wellknown/values.yaml`: comment aligned to `contextLength 12288 / maxOutputTokens 4096` vs the 16384 total serving window

The invariant now holds exactly: 12288 (advertised input) + 4096 (maxOutputTokens) = 16384 = `--max-model-len` → a catalog-honouring client can never exceed the window.

Thanks for the catch — the comment text and the value disagreed, which is precisely what you flagged.
